src/zenml/config/secret_reference_mixin.py

```diff
@@ -0,0 +1,149 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Secret reference mixin implementation."""
+from typing import TYPE_CHECKING, Any, Set
+
+from pydantic import BaseModel
+
+from zenml.logger import get_logger
+from zenml.utils import secret_utils
+
+logger = get_logger(__name__)
+
+
+class SecretReferenceMixin(BaseModel):
```

Comment:
I think the stack components have mostly the same logic somewhere, should we also refactor those to inherit from this mixin instead then?

Reply:
Yes absolutely!